src/pybind/mgr/volumes/fs/operations/volume.py

```diff
@@ -135,10 +135,23 @@ def delete_volume(mgr, volname, metadata_pool, data_pools):
     result_str = "metadata pool: {0} data pool: {1} removed".format(metadata_pool, str(data_pools))
     return r, result_str, ""
 
+def _perform_checks_before_renaming_vol(mgr, volname):
+    mdsmap = get_mds_map(mgr, volname)
+    if mdsmap is not None and len(mdsmap['up']) > 0:
+        return -errno.EPERM, (f'Volume "{volname}" has not been failed. A '
+                               'volume must be offline before attempting to '
+                               'rename it. See `ceph fs fail`.')
+    return 0, ''
+
 def rename_volume(mgr, volname: str, newvolname: str) -> Tuple[int, str, str]:
     """
     rename volume (orch MDS service, file system, pools)
     """
+    errno, errmsg = _perform_checks_before_renaming_vol(mgr, volname)
+    if errno != 0:
+        log.info(f'errno: {errno}. errmsg: {errmsg}')
```

Comment:
shouldn't this be something like `ret, status = _perform_checks_before_renaming_vol(mgr, volname)` since it can return zero too which isn't an errno, also `log.debug()` or `log.error()` suits better IMO

Reply:
In my understanding, `log.debug()` is highest level of logging meant for info that is essential only for debugging, not otherwise. And log.error is for a case were.  `log.error()` is meant for logging in cases where an error has occured within the codebase. Neither is suited here because we are generating an error by rejecting request for renaming of FS.